src/missing_tokens.py

```diff
@@ -28,67 +27,75 @@ def __init__(self, address: Address):
-            return [Address(row["token"]) for row in csv.DictReader(csv_file)]
-
-    print("Getting missing tokens from: https://dune.com/queries/236085")
-    query = DuneQuery.from_environment(
-        raw_sql=open_query("./queries/missing-tokens.sql"),
-        name=f"Missing Tokens",
-        network=Network.MAINNET,
-        parameters=[],
+
+def fetch_missing_tokens_api(dune: DuneClient) -> MissingTokenResults:
+    print(f"Fetching V1 missing tokens from {V1_QUERY.url()}")
+    v1_missing = dune.refresh(V1_QUERY)
+    print(f"Fetching V2 missing tokens from {V2_QUERY.url()}")
+    v2_missing = dune.refresh(V2_QUERY)
+
+    return MissingTokenResults(
+        v1=[Address(row["token"]) for row in v1_missing],
+        v2=[Address(row["token"]) for row in v2_missing],
     )
-    results = dune.fetch(query)
-    return [Address(row["token"]) for row in results]
 
 
 if __name__ == "__main__":
     load_dotenv()
-    parser = argparse.ArgumentParser("Missing Tokens")
-    parser.add_argument(
-        "--token-file",
-        type=str,
-        help="CSV File to read token data from (this is for Dune V2)",
-    )
-    args = parser.parse_args()
-
-    token_file = File(args.token_file) if args.token_file else None
-    # We only supply a token file for Dune V2.
-    dune_version = DuneVersion.V1 if not token_file else DuneVersion.V2
 
     w3 = Web3(
         Web3.HTTPProvider(f"https://mainnet.infura.io/v3/{os.environ['INFURA_KEY']}")
     )
-    dune_conn = DuneAPI.new_from_environment()
-    missing_tokens = fetch_missing_tokens(dune_conn, file=token_file)
-    if missing_tokens:
-        print(f"Found {len(missing_tokens)} missing tokens, fetching metadata...\n")
-        token_details = []
-        for t in missing_tokens:
+    missing_tokens = fetch_missing_tokens_api(DuneClient(os.environ["DUNE_API_KEY"]))
+
+    if not missing_tokens.is_empty():
+        print(
+            f"Found {len(missing_tokens.v1)} missing tokens on V1 "
+            f"and {len(missing_tokens.v2)} on V2. Fetching token details...\n"
+        )
+        token_details = {}
+        for token in missing_tokens.all_tokens():
             try:
                 # TODO batch the eth_calls used to construct the token contracts.
-                token_details.append(TokenDetails(address=t).to_str(dune_version))
+                token_details[token] = TokenDetails(address=token)
             except web3.exceptions.BadFunctionCallOutput as err:
-                print(f"Something wrong with token {t} - skipping.")
-        print("\n".join(token_details))
+                print(f"Something wrong with token {token} - skipping.")
+
+        v1_results = "\n".join(token_details[t].v1_string() for t in missing_tokens.v1)
+        v2_results = "\n".join(token_details[t].v2_string() for t in missing_tokens.v2)
+
+        print(f"V1 results:\n\n{v1_results}\n")
+        print(f"V2 results:\n\n{v2_results}\n")
+        # TODO - write to file!
```

Comment:
There's still this TODO in here. Just mentioning this in case you wanted to do it and forgot.

Reply:
Actually I think I want to change the todo to be insert directly in spellbook abstraction file (which would be a separate task). For write to file I guess we could sneak it into this PR.

Reply:
This later.